packages/evm/contracts/core/modules/SignaturesValidator.sol

```diff
@@ -84,41 +83,20 @@ contract SignaturesValidator is IERC712, TimeHelpers {
     }
 
     /**
-    * @dev Auth deadline encoded in calldata
-    */
-    function _deadline() internal pure returns (uint256) {
-        return _decodeExtraCalldataWord(0);
-    }
-
-    /**
-    * @dev Signature encoded in calldata
-    */
-    function _signature() internal pure returns (uint8 v, bytes32 r, bytes32 s) {
-        v = uint8(_decodeExtraCalldataWord(0x20));
-        r = bytes32(_decodeExtraCalldataWord(0x40));
-        s = bytes32(_decodeExtraCalldataWord(0x60));
-    }
-
-    /**
-    * @dev Decode original calldata
-    */
-    function _calldata() internal pure returns (bytes memory result) {
-        result = msg.data;
-        if (result.length > EXTRA_CALLDATA_LENGTH) {
-            assembly { mstore(result, sub(calldatasize, EXTRA_CALLDATA_LENGTH)) }
-        }
-    }
-
-    /**
-    * @dev Decode word from extra calldata
+    * @dev Decode extra calldata
     */
-    function _decodeExtraCalldataWord(uint256 _offset) internal pure returns (uint256 result) {
-        uint256 offset = _offset;
-        assembly {
-            let ptr := mload(0x40)
-            mstore(0x40, add(ptr, 0x20))
-            calldatacopy(ptr, sub(calldatasize, sub(EXTRA_CALLDATA_LENGTH, offset)), 0x20)
-            result := mload(ptr)
+    function _decodeCalldata() internal pure returns (bytes memory data, uint256 deadline, bytes32 r, bytes32 s, uint8 v) {
+        data = msg.data;
+        if (data.length > EXTRA_CALLDATA_LENGTH) {
+            assembly {
+                let realCalldataSize := sub(calldatasize, EXTRA_CALLDATA_LENGTH)
+                let extraCalldataPtr := add(add(data, 0x20), realCalldataSize)
+                deadline := mload(extraCalldataPtr)
+                r := mload(add(extraCalldataPtr, 0x20))
+                s := mload(add(extraCalldataPtr, 0x40))
+                v := byte(0, mload(add(extraCalldataPtr, 0x60)))
```

Comment:
Oh nice, never knew about the `byte()` helper in assembly!

Wondering if it'd be too sketchy to put `v` first, since that can be quite common and the ABI-encoding of this would match `uint8 v, bytes32 r, bytes32 s` found elsewhere. We would then be offsetting the rest by `0x21` and `0x41`?